game-core/src/test/java/games/strategy/engine/message/RemoteActionCodeTest.java

```diff
@@ -0,0 +1,49 @@
+package games.strategy.engine.message;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+
+import java.lang.reflect.Method;
+import java.util.stream.IntStream;
+import org.junit.jupiter.api.extension.ParameterContext;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.aggregator.AggregateWith;
+import org.junit.jupiter.params.aggregator.ArgumentsAccessor;
+import org.junit.jupiter.params.aggregator.ArgumentsAggregationException;
+import org.junit.jupiter.params.aggregator.ArgumentsAggregator;
+import org.junit.jupiter.params.provider.CsvFileSource;
+
+public class RemoteActionCodeTest {
+
+  @ParameterizedTest
+  @CsvFileSource(resources = "/required-op-codes.txt")
+  void verifyCorrectOpCode(
+      final int opCode, @AggregateWith(MethodAggregator.class) final Method method) {
+    final var remoteActionCode = method.getAnnotation(RemoteActionCode.class);
+
+    assertThat("No annotation present for " + method, remoteActionCode, is(notNullValue()));
+
+    assertThat("Invalid value for " + method, remoteActionCode.value(), is(opCode));
+  }
+
+  static class MethodAggregator implements ArgumentsAggregator {
+    @Override
+    public Method aggregateArguments(
+        final ArgumentsAccessor arguments, final ParameterContext context)
+        throws ArgumentsAggregationException {
+      final int offset = context.getIndex();
+      final Class<?> clazz = arguments.get(offset, Class.class);
```

Comment:
What kind of class is this expected to be? Is it an annotated one with the remote method number? Perhaps would be more clear as something like `RemoteMethodAnnotatedClass`

Reply:
Classes are never annotated with this annotation, only methods of interfaces are.
The class in question is an arbitrary subinterface of `IRemote` or `IChannelSubscriber`, containing methods that are annotated with the annotation.
I originally wanted to call it `interface`, but that's a reserved keyword too.

Reply:
Perhaps rename to  `InterfaceWithRemoteMethods`?